### PR TITLE
API End Points PyTest

### PR DESCRIPTION
**Pull request by mikecarrier4**
A file to run automatic tests on Fast API Endpoints

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by mikecarrier4** [comment]
Screen Shot 2021-09-20 at 11 59 52 AM

**Comment by mikecarrier4** [comment]
Failed API Tests are about Bot.  Will review once new Bot Pull request is merged

File: testing/test_main.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+"""Test all Posts"""
+
+
+def test_form_in():
+    response = client.post(
+        '/form-in/',
+        json={
+                "city": "cHicago",
+                "state": "ILLInoIs",
```
Comment (ahmedseragcodes) [inline testing/test_main.py:55]: Not sure what happened with location here, but I don't think case sensitivity will play an issue.

**Review by ahmedseragcodes — APPROVED** [review]
Tests make sense + don't see current bugs nor increased risk of bugs => approved

File: testing/test_main.py (hunk 1 above)
Comment (mikecarrier4) [reply testing/test_main.py:55]: It was a test to make sure that case sensitivity would not be an issue in the form as the API docs do not specify case sensitivity.  However future iterations should be done with a random string generator.

Comment (ChristopherKchilton) [reply testing/test_main.py:55]: Its nice. I don't think the code base currently has any test. We need more. I approve of tests.

**Review by ChristopherKchilton — APPROVED** [review]
The code looks concise and seems to test things appropriately. As long as the tests ran I would approve these tests.

**Review by paulstgermain — CHANGES_REQUESTED** [review]
Adding a test file is a great idea - I'd love to know more! Before approving this PR, I'd like to see you record and link a Loom video walking through this PR and explaining your work. 🙂

File: testing/test_main.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+def test_get_bot_scripts():
+    response = client.get("/select-all-from-bot-scripts")
+    assert response.status_code == 200
+
+def test_frankenbert():
+    test = 'A women walked her dog down the street and cop waved to her'
+    response = client.get(f'/frankenbert/{test}')
+    assert response.status_code == 200
+    response_body = response.json()
```
Comment (iknight7000) [inline testing/test_main.py:24]: I like the addition to the unit tests, there aren't many uses for unit tests for us.

**Review by iknight7000 — APPROVED** [review]
The addition of unit tests are unique for this project. I approve.

File: testing/test_main.py (hunk 1 above)
Comment (gaben3722) [reply testing/test_main.py:55]: These tests all look great. Case sensitivity test is something small but could cause big problems if not accounted for.

**Review by gaben3722 — APPROVED** [review]

**Comment by mikecarrier4** [comment]
[https://www.youtube.com/watch?v=J7wh5iCydDI&t=358s(url)

**Comment by mikecarrier4** [comment]
https://fastapi.tiangolo.com/tutorial/testing/

**Comment by BrokenShell** [comment]
closed

**Pull request closed without merging**